src/index.ts

```diff
@@ -1,212 +1,17 @@
-    const stackdriverLevel = STACKDRIVER_LOGGING_LEVEL_CODE_TO_NAME[levelCode];
-
-    const entryMetadata: types.StackdriverEntryMetadata = {
-      resource: this.resource,
-    };
-    if (this.labels) {
-      entryMetadata.labels = this.labels;
-    }
-
-    const data: types.StackdriverData = {};
-
-    // Stackdriver Logs Viewer picks up the summary line from the `message`
-    // property of the jsonPayload.
-    // https://cloud.google.com/logging/docs/view/logs_viewer_v2#expanding.
-    //
-    // For error messages at severity 'error' and higher, Stackdriver
-    // Error Reporting will pick up error messages if the full stack trace is
-    // included in the textPayload or the message property of the jsonPayload.
-    // https://cloud.google.com/error-reporting/docs/formatting-error-messages
-    // We prefer to format messages as jsonPayload (by putting it as a message
-    // property on an object) as that works is accepted by Error Reporting in
-    // for more resource types.
-    //
-    if (metadata && metadata.stack) {
-      msg += (msg ? ' ' : '') + (metadata as types.Metadata).stack;
-      data.serviceContext = this.serviceContext;
-    }
-    data.message = this.prefix ? `[${this.prefix}] ` : '';
-    data.message += msg;
-
-    if (is.default.object(metadata)) {
-      data.metadata =
-          this.inspectMetadata ? mapValues(metadata, util.inspect) : metadata;
-
-      // If the metadata contains a httpRequest property, promote it to the
-      // entry metadata. This allows Stackdriver to use request log formatting.
-      // https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#HttpRequest
-      // Note that the httpRequest field must properly validate as HttpRequest
-      // proto message, or the log entry would be rejected by the API. We no do
-      // validation here.
-      if (metadata.httpRequest) {
-        entryMetadata.httpRequest = metadata.httpRequest;
-        delete data.metadata!.httpRequest;
-      }
-
-      // If the metadata contains a labels property, promote it to the entry
-      // metadata.
-      // https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry
-      if (metadata.labels) {
-        entryMetadata.labels = (entryMetadata.labels) ?
-            Object.assign({}, entryMetadata.labels, metadata.labels) :
-            metadata.labels;
-        delete data.metadata!.labels;
-      }
-    }
+require('winston');
+const winstonVersion = require('winston/package.json').version;
 
-    if (metadata && metadata[LOGGING_TRACE_KEY]) {
-      entryMetadata.trace = metadata[LOGGING_TRACE_KEY];
-      delete data.metadata![LOGGING_TRACE_KEY];
-    } else {
-      const trace = getCurrentTraceFromAgent();
-      if (trace) {
-        entryMetadata.trace = trace;
-      }
-    }
+// This is a class.
+// tslint:disable-next-line:variable-name
+export const LoggingWinston: types.Logger = semver.lt(winstonVersion, '3.0.0') ?
+    require('./winston2').LoggingWinston :
```

Comment:
It looks like this is causing compilation problems for dependent TypeScript code.  In particular, when upgrading to the logging-winston in this PR in a TypeScript app, I get the following compilation error:
```
error TS2351: Cannot use 'new' with an expression whose type lacks a call or construct signature.

45   transports: [new winston.transports.Console(), new LoggingWinston()]
```

Reply:
fixed!